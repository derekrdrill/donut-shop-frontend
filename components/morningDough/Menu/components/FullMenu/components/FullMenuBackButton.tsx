import * as React from 'react';
import { Button, Grid } from '@mui/material';
import KeyboardDoubleArrowLeftIcon from '@mui/icons-material/KeyboardDoubleArrowLeft';

type FullMenuBackButtonProps = {
  selectedCategory: string | null;
  setSelectedCategory: React.Dispatch<React.SetStateAction<string | null>>;
  setSelectedCategoryName: React.Dispatch<React.SetStateAction<string | null>>;
};

const FullMenuBackButton = ({
  selectedCategory,
  setSelectedCategory,
  setSelectedCategoryName,
}: FullMenuBackButtonProps) => (
  <Grid container data-testid='full-menu-back-button'>
    {selectedCategory && (
      <Grid data-testid='selected-category' container>
        <Grid item xs={12} md={3} lg={2}>
          <Button
            color='secondary'
            onClick={
              /* istanbul ignore next */
              () => {
                setSelectedCategory(null);
                setSelectedCategoryName(null);
              }
            }
            startIcon={<KeyboardDoubleArrowLeftIcon />}
            variant='text'
          >
            Go Back
          </Button>
        </Grid>
      </Grid>
    )}
  </Grid>
);

export default FullMenuBackButton;
