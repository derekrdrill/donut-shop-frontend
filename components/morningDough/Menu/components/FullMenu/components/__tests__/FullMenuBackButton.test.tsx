import '@testing-library/jest-dom';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import FullMenuBackButton from '../FullMenuBackButton';

describe('FullMenuHead unit tests', () => {
  it('renders as expected', () => {
    const result = renderer
      .create(
        <FullMenuBackButton
          selectedCategory='hotDrinks'
          setSelectedCategory={jest.fn()}
          setSelectedCategoryName={jest.fn()}
        />,
      )
      .toJSON();
    expect(result).toMatchSnapshot();
  });

  it('renders FullMenuBackButton with selectedCategory', async () => {
    const { getByTestId } = render(
      <FullMenuBackButton
        selectedCategory={'hotDrinks'}
        setSelectedCategory={jest.fn()}
        setSelectedCategoryName={jest.fn()}
      />,
    );

    expect(getByTestId('selected-category')).toBeInTheDocument();
  });
});
