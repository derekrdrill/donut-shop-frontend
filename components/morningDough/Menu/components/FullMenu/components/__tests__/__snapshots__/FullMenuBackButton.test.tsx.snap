// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullMenuHead unit tests renders as expected 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  data-testid="full-menu-back-button"
>
  <div
    className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    data-testid="selected-category"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 MuiGrid-grid-lg-2 css-57ri8o-MuiGrid-root"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium css-1d3f8j8-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="KeyboardDoubleArrowLeftIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17.59 18 19 16.59 14.42 12 19 7.41 17.59 6l-6 6z"
            />
            <path
              d="m11 18 1.41-1.41L7.83 12l4.58-4.59L11 6l-6 6z"
            />
          </svg>
        </span>
        Go Back
      </button>
    </div>
  </div>
</div>
`;
